packages/web3torrent/src/library/paid-streaming-extension.ts

```diff
@@ -175,15 +178,20 @@ export abstract class PaidStreamingExtension implements Extension {
     wire._onRequest = function(index, offset, length) {
       log(`_onRequest: ${index}`);
 
-      messageBus.emit(PaidStreamingExtensionEvents.REQUEST, index, length, function(allow = false) {
-        if (allow) {
-          _onRequest.apply(wire, [index, offset, length]);
-          log(`_onRequest PASS - ${index}`);
-        } else {
-          wire._onCancel(index, offset, length);
-          log(`_onRequest CHOKED - ${index}`);
-        }
-      });
+      if (this.paidStreamingExtension.isForceChoking) {
+        wire._onCancel(index, offset, length);
+        log(`_onRequest AUTO CHOKE - ${index}, ${offset}, ${length}`);
```

Comment:
This makes it so that the library doesn't even get to see choked requests if it's supposed to be choking them.